src/vs/editor/contrib/hover/browser/contentHover.ts

```diff
@@ -618,10 +618,11 @@ export class ContentHoverWidget extends ResizableContentWidget {
 		if (!this._editor || !this._editor.hasModel()) {
 			return;
 		}
-		const editorBox = dom.getDomNodePagePosition(this._editor.getDomNode());
-		const glyphMarginWidth = this._editor.getLayoutInfo().glyphMarginWidth;
-		const leftOfContainer = this._hover.containerDomNode.offsetLeft;
-		return editorBox.width + editorBox.left - leftOfContainer - glyphMarginWidth;
+		const editorLayoutInfo = this._editor.getLayoutInfo();
+		const glyphMarginRight = editorLayoutInfo.glyphMarginLeft + editorLayoutInfo.glyphMarginWidth;
+		const cursorPosition = this._editor._getViewModel().getPrimaryCursorState().viewState.position;
+		const bodyBoxWidth = dom.getClientArea(document.body).width;
+		return bodyBoxWidth - glyphMarginRight - this._editor.getOffsetForColumn(cursorPosition.lineNumber, cursorPosition.column);
```

Comment:
Works much better and is much less code 👏 Tho, now that I see and against my advise I wonder if we should ignore the cursor position all together. Because you are now penalised for being a split editor on the RHS or being far indented. Maybe we allow the width to be the whole body-width minus the usual paddings. The suggest widget (contrary to its details widget) is that here: 

https://github.com/microsoft/vscode/blob/0a6b7a310347de787da1034038442802f3ac0194/src/vs/editor/contrib/suggest/browser/suggestWidget.ts#L785

Reply:
I'll change it to the width you specified